web-ui/src/main/resources/catalog/components/admin/uiconfig/UiConfigDirective.js

```diff
@@ -42,23 +43,122 @@
 
           var testAppUrl = '../../catalog/views/api/?config=';
 
-          function init() {
-            // merge on top of default config
-            scope.jsonConfig = angular.merge(
-              // If the config is empty, use the default one set in CatController
-              (Object.keys(scope.config).length === 0 || scope.config === "null" || !scope.config ?
-                gnGlobalSettings.getDefaultConfig() :
-                gnGlobalSettings.getMergeableDefaultConfig()),
-              angular.fromJson(scope.config));
+          scope.optionsToAdd = undefined;
+
+          var preferredKey = [
+            'mods.header.languages',
+            'mods.home.facetConfig',
+            'mods.search.facetConfig',
+            'mods.editor.facetConfig',
+            'mods.search.filters'
+          ];
+
+          function buildLabel(tokens) {
+            var prefix = $translate.instant(
+              'ui-'
+              + (tokens[0] === 'mods' ? 'mod' : tokens[0])
+              + (tokens.length > 1 ? '-' + tokens[1] : ''));
+            return prefix
+              + (tokens.length > 2
+                ? ' / ' + $translate.instant('ui-' + tokens[tokens.length -1])
+                : '');
+          }
+          function collectConfigOption(jsonConfig) {
```

Comment:
Same as on the other file, this has no documentation and no clear purpose; I expect maintenance of this code to be interesting